### PR TITLE
Fix warnings from black, isort and mypy

### PR DESCRIPTION
**Pull request by hoh**
Depends on https://github.com/aleph-im/aleph-vm/pull/325

This fixes all basic warnings reported by `black` `isort` and `mypy` and uses `pathlib.Path` more generally.

### CODE REVIEW

File: vm_supervisor/utils.py [hunk 1]
```diff
@@ -52,3 +53,23 @@ async def run_and_log_exception(coro: Coroutine):
 def create_task_log_exceptions(coro: Coroutine, *, name=None):
     """Ensure that exceptions running in coroutines are logged."""
     return asyncio.create_task(run_and_log_exception(coro), name=name)
+
+
+async def run_in_subprocess(
+    command: List[str], check: bool = True, stdin_input: Optional[bytes] = None
```
Comment (nesitor) [inline vm_supervisor/utils.py:59]: The encoding parameter is needed for some commands related with devmapper

File: vm_supervisor/vm/firecracker/program.py [hunk 2]
```diff
@@ -297,7 +299,11 @@ async def configure(self) -> None:
             resources=self.resources, drives=self.fvm.drives
         )
         interface: Interface = Interface.from_entrypoint(self.resources.code_entrypoint)
-        input_data: Optional[bytes] = read_input_data(self.resources.data_path)
+        # input_data: Optional[bytes] = read_input_data(self.resources.data_path)
```
Comment (nesitor) [inline vm_supervisor/vm/firecracker/program.py:302]: This comment should be removed?

File: vm_supervisor/storage.py [hunk 3]
```diff
@@ -188,44 +195,34 @@ async def create_loopback_device(path: Path, read_only: bool = False) -> str:
     if read_only:
         command_args.append("--read-only")
     command_args.append(str(path))
-    loop_device = subprocess.run(
-        command_args, check=True, capture_output=True, encoding="UTF-8"
-    ).stdout.strip()
+    stdout = await run_in_subprocess(command_args)
+    loop_device = stdout.strip().decode()
```
Comment (nesitor) [inline vm_supervisor/storage.py:199]: We need to decode to utf-8 here also

File: vm_supervisor/utils.py (hunk 1 above)
Comment (hoh) [reply vm_supervisor/utils.py:59]: Stdin and stdout are always expected to be bytes. Passing an extra parameter to ask to `encode()` / `decode()` would not make the code clearer or easier to manipulate.

**Pull request merged**